### PR TITLE
fix(oas): add missing x-codegen + fix schema naming and $ref

### PR DESCRIPTION
**Pull request by patrick-medusajs**
## What

Add missing x-codegen + fix schema naming and $ref 

## Why

OAS code generator was missing information to be able to match current JS client.

## How

Ensure OAS matches JS client methods.

### CODE REVIEW

File: packages/medusa/src/api/routes/admin/inventory-items/get-inventory-item.ts [hunk 1]
```diff
@@ -13,6 +13,9 @@ import { joinLevels } from "./utils/join-levels"
  *   - (path) id=* {string} The ID of the Inventory Item.
  *   - (query) expand {string} Comma separated list of relations to include in the results.
  *   - (query) fields {string} Comma separated list of fields to include in the results.
+ * x-codegen:
+ *   method: retrieve
+ *   queryParams: AdminGetInventoryItemsItemParams
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/inventory-items/get-inventory-item.ts:18]: https://github.com/medusajs/medusa/blob/194f5e5a70bba17a33d50cc32cec486bc0299041/packages/medusa-js/src/resources/admin/inventory-item.ts#L25-L29

File: packages/medusa/src/api/routes/admin/inventory-items/list-inventory-items.ts [hunk 2]
```diff
@@ -51,6 +51,9 @@ import { AdminInventoryItemsListWithVariantsAndLocationLevelsRes } from "."
  *   - (query) height {string} height to search for.
  *   - (query) width {string} width to search for.
  *   - (query) requires_shipping {string} requires_shipping to search for.
+ * x-codegen:
+ *   method: list
+ *   queryParams: AdminGetInventoryItemsParams
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/inventory-items/list-inventory-items.ts:56]: https://github.com/medusajs/medusa/blob/194f5e5a70bba17a33d50cc32cec486bc0299041/packages/medusa-js/src/resources/admin/inventory-item.ts#L85-L88

File: packages/medusa/src/api/routes/admin/inventory-items/list-location-levels.ts [hunk 3]
```diff
@@ -15,6 +15,9 @@ import { FindParams } from "../../../../types/common"
  *   - (query) limit=20 {integer} Limit the number of stock locations levels returned.
  *   - (query) expand {string} Comma separated list of relations to include in the results.
  *   - (query) fields {string} Comma separated list of fields to include in the results.
+ * x-codegen:
+ *   method: listLocationLevels
+ *   queryParams: AdminGetInventoryItemsItemLocationLevelsParams
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/inventory-items/list-location-levels.ts:20]: https://github.com/medusajs/medusa/blob/194f5e5a70bba17a33d50cc32cec486bc0299041/packages/medusa-js/src/resources/admin/inventory-item.ts#L176-L180

File: packages/medusa/src/api/routes/admin/reservations/list-reservations.ts [hunk 4]
```diff
@@ -63,6 +63,9 @@ import {
  *   - (query) limit=20 {integer} Limit the number of Reservations returned.
  *   - (query) expand {string} (Comma separated) Which fields should be expanded in the product category.
  *   - (query) fields {string} (Comma separated) Which fields should be included in the product category.
+ * x-codegen:
+ *   method: list
+ *   queryParams: AdminGetReservationsParams
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/reservations/list-reservations.ts:68]: https://github.com/medusajs/medusa/blob/194f5e5a70bba17a33d50cc32cec486bc0299041/packages/medusa-js/src/resources/admin/reservations.ts#L36-L39

File: packages/medusa/src/api/routes/admin/reservations/index.ts [hunk 5]
```diff
@@ -56,7 +56,7 @@ export default (app) => {
 }
 
 /**
- * @schema AdminPostReservationsReq
+ * @schema AdminReservationsRes
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/reservations/index.ts:59]: note: schema name was not matching type name

File: packages/medusa/src/api/routes/admin/reservations/index.ts [hunk 6]
```diff
@@ -69,7 +69,7 @@ export type AdminReservationsRes = {
 }
 
 /**
- * @schema AdminGetReservationReservationsReq
+ * @schema AdminReservationsListRes
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/reservations/index.ts:72]: note: schema name was not matching type name

**Review by patrick-medusajs — COMMENTED** [review]
self-review: highlighting noteworthy code changes

**Review by pKorsholm — APPROVED** [review]
LGTM 😄

File: packages/medusa/src/api/routes/admin/orders/get-reservations.ts [hunk 7]
```diff
@@ -40,7 +40,7 @@ import { extendedFindParamsMixin } from "../../../../types/common"
  *     content:
  *       application/json:
  *         schema:
- *           $ref: "#/components/schemas/AdminGetReservationReservationsReq"
+ *           $ref: "#/components/schemas/AdminReservationsRes"
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/orders/get-reservations.ts:43]: This should be `AdminReservationsListRes`

**Review by shahednasser — CHANGES_REQUESTED** [review]

File: packages/medusa/src/api/routes/admin/orders/get-reservations.ts (hunk 7 above)
Comment (patrick-medusajs) [reply packages/medusa/src/api/routes/admin/orders/get-reservations.ts]: Good catch 🎣 . Done in efda8f4695e084de83426d45b33a56af704b15e4

**Review by shahednasser — APPROVED** [review]
LGTM

**Pull request merged**